server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1888,26 +1876,23 @@ static String createEmptyTranslog(
      * @param shardId                 the {@link ShardId}
      * @param initialGlobalCheckpoint the global checkpoint to initialize the translog with
      * @param primaryTerm             the shard's primary term to initialize the translog with
-     * @param translogUUID            the unique identifier to initialize the translog with
-     * @param factory                 a {@link ChannelFactory} used to open translog files
+     * @param channelFactory          a {@link ChannelFactory} used to open translog files
      * @return the translog's unique identifier
      * @throws IOException if something went wrong during translog creation
      */
-    public static String createEmptyTranslog(
-        final Path location,
-        final ShardId shardId,
-        final long initialGlobalCheckpoint,
-        final long primaryTerm,
-        @Nullable final String translogUUID,
-        @Nullable final ChannelFactory factory
+    static String createEmptyTranslog(
+        Path location,
+        long initialGlobalCheckpoint,
+        ShardId shardId,
+        ChannelFactory channelFactory,
+        long primaryTerm
     ) throws IOException {
         IOUtils.rm(location);
         Files.createDirectories(location);
 
         final long generation = 1L;
         final long minTranslogGeneration = 1L;
-        final ChannelFactory channelFactory = factory != null ? factory : FileChannel::open;
-        final String uuid = Strings.hasLength(translogUUID) ? translogUUID : UUIDs.randomBase64UUID();
+        final String uuid = UUIDs.randomBase64UUID();
```

Comment:
I think there is one usage where we pass a non-null translog UUID here

Reply:
You're right, left this unchanged and only left the one useless method removal in this PR now. Sorry for the noise here.